sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StreamingWriteTables.java

```diff
@@ -339,14 +402,17 @@ public WriteResult expand(PCollection<KV<TableDestination, ElementT>> input) {
           input
               .apply("ShardTableWrites", ParDo.of(new GenerateShardedTable<>(numShards)))
               .setCoder(KvCoder.of(ShardedKeyCoder.of(StringUtf8Coder.of()), elementCoder))
-              .apply("TagWithUniqueIds", ParDo.of(new TagWithUniqueIds<>()))
+              .apply("TagWithUniqueIds", ParDo.of(new TagWithUniqueIds<>(toUniqueId)))
               .setCoder(
                   KvCoder.of(
                       ShardedKeyCoder.of(StringUtf8Coder.of()),
                       TableRowInfoCoder.of(elementCoder)));
 
+      if (!skipReshuffle) {
+        shardedTagged = shardedTagged.apply(Reshuffle.of());
```

Comment:
My preference would be to require the unique id function if we skip the reshuffle, otherwise we get duplicate elements, right? Prefer to have no codepath that is easily incorrect. Is there a specific reason this doesn't work?

Reply:
Changed as discussed to withDeterministicRecordIdFn